### PR TITLE
Adds flag to check for Submit/Receive-only users on legacy table

### PR DESCRIPTION
**Pull request by dakota002**
I'll work on a script to populate this, similar to the legacy submitters. 

Also will need an update to the post sign up to remove users from this table as they sign up for Cognito accounts

### CODE REVIEW

**Review by lpsinger — COMMENTED** [review]
What about just a boolean flag on the `legacy_users` table that says whether they receive circulars?

**Comment by dakota002** [comment]
That could work too! I could use a scan with a filter on that bool to get the receivers, and I think the submitter code would need to be updated too

**Comment by lpsinger** [comment]
> I could use a scan with a filter on that bool to get the receivers, and I think the submitter code would need to be updated too

Remember that we always want to avoid scans because they are expensive both in terms of time and billing. You can add a global secondary that includes the boolean field as a partition or sort key so that you can query on it.

File: app/events/email-incoming/index.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ interface UserData {
   sub?: string
   name?: string
   affiliation?: string
+  receiveOnly?: boolean
```
Comment (lpsinger) [inline app/events/email-incoming/index.ts:35]: How about two flags, `submit` and `receive`? Users can be submit only, receive only, or submit and receive. Is that right, @jracusin?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/events/email-incoming/index.ts (hunk 1 above)
Comment (dakota002) [reply app/events/email-incoming/index.ts:35]: Sure! I can add that now

**Comment by lpsinger** [comment]
Would you please rebase this?

File: app.arc [hunk 2]
```diff
@@ -81,6 +81,10 @@ circulars_subscriptions
   sub *String
   name circularsSubscriptionsBySub
 
+legacy_users
+  receive *Number
```
Comment (lpsinger) [inline app.arc:85]: DynamoDB does have a boolean type. Does it support boolean fields as partition keys?

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — APPROVED** [review]

**Pull request merged**